### PR TITLE
Avoid prompting user to accept hostkey in OOD shell

### PR DESCRIPTION
**Pull request by sjpb**
... by adding the host's key to global known hosts.

### CODE REVIEW

File: ansible/roles/openondemand/tasks/main.yml [hunk 1]
```diff
@@ -128,3 +128,16 @@
 #   loop:
 #     - /var/www/ood/public
 #     - /usr/share/ondemand-dex/web/themes/
+
+- name: Keyscan login host
+  command:
+    cmd: "ssh-keyscan -t ssh-ed25519 {{ openondemand_clusters.slurm.v2.login.host }}"
+  register: _openondemand_login_key
+
+- name: Add login hostkey to known hosts
```
Comment (sd109) [inline ansible/roles/openondemand/tasks/main.yml:137]: Should we use line_in_file here instead of copy?

**Review by sd109 — CHANGES_REQUESTED** [review]

**Comment by sjpb** [comment]
@sd109  tested, works, is idempotent.

File: ansible/roles/openondemand/tasks/main.yml [hunk 2]
```diff
@@ -128,3 +128,19 @@
...
+    cmd: "ssh-keyscan {{ openondemand_clusters.slurm.v2.login.host }}"
+  register: _openondemand_login_key
+  changed_when: false
+
+- name: Add login hostkeys to known hosts
+  blockinfile:
+    path: /etc/ssh/ssh_known_hosts
+    create: true
+    block: "{{ _openondemand_login_key.stdout_lines | sort | join('\n') }}"
+    marker: "# {mark} ANSIBLE MANGED BLOCK: openondemand login host" # allows other tasks to use blockinfile on this file
```
Comment (sd109) [inline ansible/roles/openondemand/tasks/main.yml:143]:
```suggestion
    marker: "# {mark} ANSIBLE MANAGED BLOCK: openondemand login host" # allows other tasks to use blockinfile on this file
```

**Review by sd109 — APPROVED** [review]
LGTM

**Review by sd109 — APPROVED** [review]

**Pull request merged**